cmdstanpy/lib.py

```diff
@@ -81,194 +82,360 @@ def data_file(self) -> str:
+                        self.warmup_iters
                     )
                 )
-        if self.warmup_draws is not None:
-            if self.warmup_draws < 0:
+            for x in self.warmup_schedule:
+                if x < 0 or x > 1:
+                    raise ValueError(
+                        'warmup_schedule should be triple of precentages that'
+                        ' sum to 1, e.g. (0.1, 0.8, 0.1), found {}'.format(
+                            self.warmup_schedule
+                        )
+                    )
+
+            num_warmup = 1000
+            if self.warmup_iters is not None:
+                num_warmup = self.warmup_iters
+            self.init_buffer = math.floor(num_warmup * self.warmup_schedule[0])
+            self.term_buffer = math.floor(num_warmup * self.warmup_schedule[2])
+
+        if self.thin is not None:
+            if self.thin < 1:
                 raise ValueError(
-                    'warmup_draws must be a '
-                    'non-negative integer value'.format(
-                        self.warmup_draws
+                    'thin must be at least 1, found {}'.format(self.thin)
+                )
+
+        if self.max_treedepth is not None:
+            if self.max_treedepth < 1:
+                raise ValueError(
+                    'max_treedepth must be at least 1, found {}'.format(
+                        self.max_treedepth
                     )
                 )
-        # TODO: check type/bounds on all other controls
-        # positive int values
+
+        if self.step_size is not None:
+            if isinstance(self.step_size, (int, float)):
+                if self.step_size < 0:
+                    raise ValueError(
+                        'step_size must be > 0, found {}'.format(self.step_size)
+                        )
+            else:
+                if len(self.step_size) != len(self.chain_ids):
+                    raise ValueError(
+                        'number of step_sizes must match number of chains '
+                        ' found {} step_sizes for {} chains '.format(
+                            len(self.step_size), len(self.chain_ids)
+                        )
+                    )
+                for i in range(len(self.step_size)):
+                    if self.step_size[i] < 0:
+                        raise ValueError(
+                            'step_size must be > 0, found {}'.format(
+                                self.step_size[i]
+                            )
+                        )
+
+        if self.metric is not None:
+            dims = None
+            if isinstance(self.metric, str):
+                if self.metric == 'diag':
+                    self.metric = 'diag_e'
+                elif self.metric == 'dense':
+                    self.metric = 'dense_e'
+                else:
+                    if not os.path.exists(self.metric):
+                        raise ValueError('no such file {}'.format(self.metric))
```

Comment:
I think self.metric could also accept `diag_e` and `dense_e` kwargs

Reply:
good point - will do.

Reply:
the same goes for the inits arg as well, no?  checking there as well.